packages/editor/src/plugins/image/components/licence-dropdown.tsx

```diff
@@ -0,0 +1,111 @@
+import { EditorTooltip } from '@editor/editor-ui/editor-tooltip'
+import { faQuestionCircle } from '@fortawesome/free-regular-svg-icons'
+import { faCaretDown, faCaretUp } from '@fortawesome/free-solid-svg-icons'
+import * as Select from '@radix-ui/react-select'
+import React, { useState, useEffect, useRef } from 'react'
+
+import { FaIcon } from '@/components/fa-icon'
+import { useEditorStrings } from '@/contexts/logged-in-data-context'
+import { cn } from '@/helper/cn'
+
+interface LicenseDropdownProps {
+  defaultLicense?: string
+  onLicenseChange?: (license: string) => void
+  isPixabayImage?: boolean
+}
+
+export const LicenseDropdown: React.FC<LicenseDropdownProps> = ({
+  onLicenseChange,
+  isPixabayImage = false,
+}) => {
+  const editorStrings = useEditorStrings()
+  const imageStrings = editorStrings.plugins.image
+  const { licences, licenceHelpText } = imageStrings
+  const licenceNames = Object.values(licences)
+
+  const [selectedLicense, setSelectedLicense] = useState(licenceNames[0])
+  const [isConfirmed, setIsConfirmed] = useState(false)
+  const [isOpen, setIsOpen] = useState(false)
+  const [triggerWidth, setTriggerWidth] = useState(0)
+
+  const triggerRef = useRef<HTMLLabelElement>(null)
+
+  useEffect(() => {
+    if (isPixabayImage) {
+      const pixabayLicence =
+        licenceNames.find((name) => name.toLowerCase().includes('pixabay')) ||
+        ''
+      setSelectedLicense(pixabayLicence)
+      onLicenseChange?.(pixabayLicence)
+      setIsConfirmed(true)
+    }
+  }, [isPixabayImage, onLicenseChange, licenceNames])
+
+  useEffect(() => {
+    setTriggerWidth(triggerRef.current ? triggerRef.current.offsetWidth / 3 : 0)
+  }, [])
+
+  const handleLicenseChange = (newLicence: string) => {
+    setSelectedLicense(newLicence)
+    onLicenseChange?.(newLicence)
+    setIsConfirmed(true)
+  }
+
+  return (
+    <label
+      ref={triggerRef}
+      className="mx-auto mb-0 mt-5 flex items-center text-almost-black"
+    >
+      <span className="serlo-tooltip-trigger w-1/4">
+        Lizenz
+        <FaIcon className="ml-2" icon={faQuestionCircle} />
+        <EditorTooltip text={licenceHelpText} />
+      </span>
+      <span className="flex w-3/4 flex-row border-2 border-transparent">
+        <Select.Root
```

Comment:
I have some problems with this component. Not sure if it's the comp or the implementation.
- keyboard: I can open the options, but not change them
- opening the options changes the height of the select, causes the layout to shift